lib/interactor.rb

```diff
@@ -163,4 +163,30 @@ def call
   # Returns nothing.
   def rollback
   end
+
+  private
+
+  # Internal: Determine what arguments (if any) should be passed to the "call"
+  # instance method when invoking an Interactor. The "call" instance method may
+  # accept any combination of positional and keyword arguments. This method
+  # will extract values from the context in order to populate those arguments
+  # based on their names.
+  #
+  # Returns an Array of arguments to be applied as an argument list.
+  def arguments_for_call # rubocop:disable Metrics/MethodLength
```

Comment:
You are right, aborting execution on load would be perfect. But I guess it's possible only if we force users to `include Interactor` after defining `call`, which is inconvenient:

```ruby
module M
  def self.included(base)
    print "From #{base.name}: "
    puts base.instance_methods.grep(/call/).inspect
  end
end

class A
  include M
  def call(foo)
  end
end

class B
  def call(foo)
  end
  include M
end

# From A: []
# From B: [:call]
```

However, with runtime approach, method with wrong signature will raise an error at its very first run. I hope even those who don't write automated tests check their code manually at least *once* before deploying to production 😃